### PR TITLE
Fix #2866: Xcode 12/iOS 14 support

### LINKED ISSUE #2866: XCode 12/iOS 14 support
Make the app to work with XCode 12/iOS 14

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #2866 

Changes overview:

- Carthage deps updated
- sqlcipher is now a fat library, not build in repo anymore
- Added workaround for carthage iOS14 problems

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

This needs a full manual pass tests.

Most important items:
- Check if 'save logins' feature still works, for both new and upgraded users.
- Look for UI bugs and or animations issues, this is a common thing to break when upgrading iOS versions

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -3,9 +3,10 @@ github "rs/SDWebImage"			"4.4.6"
...
 github "agilebits/onepassword-app-extension" "add-framework-support"
 github "DaveWoodCom/XCGLogger"		"7.0.0"
-github "cezheng/Fuzi"			"3.1.0"
-github "SwiftyJSON/SwiftyJSON"		"5.0.0"
+github "cezheng/Fuzi"			"3.1.2"
+github "SwiftyJSON/SwiftyJSON" "5.0.0"
 github "facebook/pop"			"1.0.12"
 github "venmo/Static"			"v4.0.0"
-github "airbnb/lottie-ios" "3.1.8"
+github "airbnb/lottie-ios" "1692cb4d0fcf0406dcaf61bf7cafea49dca51354"
```
Comment (iccub) [inline Cartfile:10]: latest lottie release doesn't work with xcode 12, I pinned latest master commit here

**Comment by iccub** [comment]
note, this doesnt build on device atm, sqlcipher lib works on simulator only

**Comment by iccub** [comment]
Update: it works fine on the device now

File: carthage_command.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+xcodesdk=`xcodebuild -scheme Debug  -showBuildSettings  | grep -i 'SDK_VERSION =' | sed 's/[ ]*SDK_VERSION = //' | colrm 3`
```
Comment (iccub) [inline carthage_command.sh:1]: This workaround script is taken from Firefox

Comment (kylehickinson) [reply carthage_command.sh:1]: fastlane runs Carthage as well, but uses the fastlane command directly… we may have to switch it to use this script instead?

Comment (iccub) [reply carthage_command.sh:1]: good point, it had to be done, I fixed it and tested by uploading dev build to app store

File: carthage_command.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+echo Xcode SDK: "$xcodesdk"
+if [[ "$xcodesdk" != "13" ]]; then
+  echo XCode 12 version detected! ••• Please ensure this is correct. •••
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64=arm64 arm64e armv7 armv7s armv6 armv8' > /tmp/tmp.xcconfig
+echo 'EXCLUDED_ARCHS=$(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT))' >> /tmp/tmp.xcconfig
+echo 'IPHONEOS_DEPLOYMENT_TARGET=11.4' >> /tmp/tmp.xcconfig
```
Comment (kylehickinson) [inline carthage_command.sh]: Is this our own deployment target?

Comment (iccub) [reply carthage_command.sh]: taken from FF, can be removed, thanks for catching it

**Review by kylehickinson — APPROVED** [review]
Record for most deleted in one PR? 😂

**Pull request merged**